debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9ConstantPoolCommand.java

```diff
@@ -0,0 +1,507 @@
+        if (cpType == J9ConstantPool.J9CPTYPE_STRING) return  "String";
+        if (cpType == J9ConstantPool.J9CPTYPE_INT) return  "Int";
+        if (cpType == J9ConstantPool.J9CPTYPE_FLOAT) return  "Float";
+        if (cpType == J9ConstantPool.J9CPTYPE_LONG) return  "Long";
+        if (cpType == J9ConstantPool.J9CPTYPE_DOUBLE) return  "Double";
+        if (cpType == J9ConstantPool.J9CPTYPE_FIELD) return  "Fieldref";
+        if (cpType == J9ConstantPool.J9CPTYPE_INSTANCE_METHOD) return  "Methodref";
+        if (cpType == J9ConstantPool.J9CPTYPE_STATIC_METHOD) return  "Methodref";
+        if (cpType == J9ConstantPool.J9CPTYPE_INTERFACE_STATIC_METHOD) return  "Methodref";
+        if (cpType == J9ConstantPool.J9CPTYPE_INTERFACE_INSTANCE_METHOD) return  "Methodref";
+        if (cpType == J9ConstantPool.J9CPTYPE_HANDLE_METHOD) return  "HandleMethodref";
+        if (cpType == J9ConstantPool.J9CPTYPE_INTERFACE_METHOD) return  "InterfaceMethodref";
+        if (cpType == J9ConstantPool.J9CPTYPE_METHOD_TYPE) return  "MethodType";
+        if (cpType == J9ConstantPool.J9CPTYPE_METHODHANDLE) return  "MethodHandle";
+        if (cpType == J9ConstantPool.J9CPTYPE_ANNOTATION_UTF8) return  "UTF8";
+        if (cpType == J9ConstantPool.J9CPTYPE_CONSTANT_DYNAMIC) return  "ConstantDynamic";
+        return "Incorrect or unsupported type";
+    }
+
+    private J9ClassPointer findClassByName(J9JavaVMPointer vm, String name) throws CorruptDataException {
+        ClassSegmentIterator iterator = new ClassSegmentIterator(vm.classMemorySegments());
+        while (iterator.hasNext()) {
+            J9ClassPointer classRef = (J9ClassPointer) iterator.next();
+            String className = null;
+            if (J9ClassHelper.isArrayClass(classRef)) {
+                className = J9ClassHelper.getArrayName(classRef);
+            } else {
+                className = J9ClassHelper.getName(classRef);
+            }
+            if (className.equals(name)) {
+                return classRef;
+            }
+        }
+        return null;
+    }
+
+    private boolean sameNameAndSignature(J9ROMNameAndSignaturePointer a, J9ROMNameAndSignaturePointer b) throws CorruptDataException {
+        return J9UTF8Helper.stringValue(a.name()).equals(J9UTF8Helper.stringValue(b.name())) &&
+            J9UTF8Helper.stringValue(a.signature()).equals(J9UTF8Helper.stringValue(b.signature()));
+    }
+
+    private boolean areAnyBitsSet(long value, long mask) {
+        return (value & mask) != 0;
+    }
+
+    private J9RAMClassRefPointer J9VMConstantPoolClassRefAt(J9JavaVMPointer vm, long index) throws CorruptDataException {
+        J9RAMConstantPoolItemPointer jclConstantPool = vm.jclConstantPoolEA();
+        J9RAMClassRefPointer ref = J9RAMClassRefPointer.cast(jclConstantPool.add(index));
+        return ref;
+    }
+
+    private long parseHexString(String hexValue) {
+        return Long.parseLong(hexValue.substring(2), 16);
+    }
```

Comment:
What could be the solution? `longValue()` actually returns the address rather than the content so that's the only way I found to get the content of a pointer.

Reply:
For example, in the case of `J9CPTYPE_LONG`
```
	value = Long.toString(parseHexString(I64Pointer.cast(romCPP).getHexValue()));
```
could be simplified to
```
	value = Long.toString(romCPP.getAddress());
```
but, as you say, you might want to dereference the pointer to get the value
```
	value = Long.toString(I64Pointer.cast(romCPP).at(0));
```